src/Graph.ts

```diff
@@ -0,0 +1,174 @@
+/**
+ * Transform array to graph
+ */
+
+/**
+ * TYPES
+ */
+
+type node = {
+  x: Number;
+  y: Number;
+  value: Number;
+  isExit: Boolean;
+};
+
+type edge = {
+  from: string;
+  to: string;
+  direction: string;
+};
+
+type Graph = {
+  nodes: {
+    [coordinates: string]: node;
+  };
+  edges: Array<edge>;
+};
+
+/**
+ * FUNCTIONS
+ */
+
+export function boardToGraph(board: Array<Array<Number>>): Graph | undefined {
+
+  if (board.length < 1) return undefined;
+
+  let graphFromBoard: Graph = newBlankGraph();
+
+
+  const verticalLines: Number = board.length;
+    const horizontalLines: Number = board[0].length;
+  
+
+    for (let hIndex = 0; hIndex < horizontalLines; hIndex++){
+        for (let vIndex = 0; vIndex < verticalLines; vIndex++){
+
+            graphFromBoard = addBoardBoundaries(graphFromBoard, board, hIndex, vIndex)
+            graphFromBoard = addDirections(graphFromBoard, board, hIndex, vIndex)
+
+        }
+    }
+
+  return graphFromBoard;
+
+}
+
+function newBlankGraph(): Graph {
+  return {
+    nodes: {},
+    edges: [],
+  };
+}
+
+function addBoardBoundaries(graph: Graph, board: Array<Array<Number>>, hIndex: number, vIndex: number): Graph {
```

Comment:
I don't understand this function name. You're adding one board location, I don't see anything with boundaries.

Reply:
Function Name has been replaced with fillNodes, which is, i admit, more meaningful. In the [last commit](https://github.com/marmelab/kuba-integration/commit/52c48e332abaf5397cf6debd1512e2111062ba31)